### PR TITLE
Add support for PaLM API v1beta3 version

### PR DESCRIPTION
**Pull request by antas-marcin**
### Review checklist

- [ ] Documentation has been updated, if necessary. Link to changed documentation:
- [ ] Chaos pipeline run or not necessary. Link to pipeline:
- [ ] All new code is covered by tests where it is reasonable.
- [ ] Performance tests have been run or not necessary.

### CODE REVIEW

File: modules/generative-palm/clients/palm.go [hunk 1]
```diff
@@ -128,34 +125,115 @@ func (v *palm) Generate(ctx context.Context, cfg moduletools.ClassConfig, prompt
...
+		return nil, errors.Wrap(err, "unmarshal response body")
+	}
+
+	if err := v.checkResponse(statusCode, resBody.Error); err != nil {
+		return nil, err
+	}
+
+	if len(resBody.Candidates) > 0 {
+		return v.getGenerateResponse(resBody.Candidates[0].Content)
+	}
```
Comment (parkerduckworth) [inline modules/generative-palm/clients/palm.go:146]: When would the response body contain more than one candidate (if ever)? If it does, are we not concerned with the contents?

File: modules/generative-palm/clients/palm.go [hunk 2]
```diff
@@ -128,34 +125,115 @@ func (v *palm) Generate(ctx context.Context, cfg moduletools.ClassConfig, prompt
...
+				Messages: []generateMessage{
+					{
+						Content: prompt,
+					},
+				},
+			},
+			Temperature:    settings.Temperature(),
+			TopP:           settings.TopP(),
+			TopK:           settings.TopK(),
+			CandidateCount: 1,
```
Comment (parkerduckworth) [inline modules/generative-palm/clients/palm.go:213]: This probably answers my above question 🙂

**Review by parkerduckworth — APPROVED** [review]

File: modules/generative-palm/clients/palm.go (hunk 1 above)
Comment (antas-marcin) [reply modules/generative-palm/clients/palm.go:146]: There will be only 1 or 0 candidates. During the `generative request` I'm setting `CandidateCount: 1` (which is a default setting).

File: modules/generative-palm/clients/palm.go (hunk 2 above)
Comment (antas-marcin) [reply modules/generative-palm/clients/palm.go:213]: yes! :)

**Pull request merged**